sync/pending_polling.go

```diff
@@ -0,0 +1,673 @@
+package sync
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/NethermindEth/juno/blockchain"
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/feed"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/ethereum/go-ethereum/common/lru"
+)
+
+const (
+	preLatestCacheSize = 10
+	blockHashLag       = 10
+)
+
+// handleTickerPreLatest polls a pre-latest once and either:
+//   - emits it to out and returns true when if delivered,
+//   - caches it by its ParentHash for a future head and returns false,
+//   - returns false on errors or context cancellation.
+//
+// Caller should invoke only when at tip and not yet delivered for the current head.
+func (s *Synchronizer) handleTickerPreLatest(
+	ctx context.Context,
+	currentHead *core.Block,
+	seenByParent *lru.BasicLRU[felt.Felt, *core.PreLatest],
```

Comment:
This cache is known to be very, very inefficient if used wrong. Just a note that we are making the most of it.